lib/http-client.js

```diff
@@ -1,6 +1,14 @@
 const axios = require('axios');
 const { logRequestError, logRequestInfo, logResponseInfo } = require('./logging');
 
+const paramCount = {
+  delete: 1,
+  get: 1,
+  patch: 2,
+  post: 2,
+  put: 2,
```

Comment:
Can this just be `configIndexes` instead so we're not subtracting 1 below?

Reply:
I see this as cargo'ing conditional complexity around using surprising data structures. I would rather have a few intentional methods than the dynamic "foreach verb" approach, even if it is two categories...